partner_contact_department/models/res_partner.py

```diff
@@ -0,0 +1,28 @@
+# © 2014-2015 Tecnativa S.L. - Jairo Llopis
+# © 2016 Tecnativa S.L. - Vicent Cubells
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    department_id = fields.Many2one("res.partner.department", "Department")
+
+
+class ResPartnerDepartment(models.Model):
+    _name = "res.partner.department"
+    _order = "parent_path"
+    _parent_order = "name"
+    _parent_store = True
+    _description = "Department"
+
+    name = fields.Char(required=True, translate=True)
+    parent_id = fields.Many2one(
+        "res.partner.department", "Parent department", ondelete="restrict"
+    )
+    child_ids = fields.One2many(
+        "res.partner.department", "parent_id", "Child departments"
+    )
+    parent_path = fields.Char(index=True, unaccent=False)
```

Comment:
it would be necessary to review the recursion
https://github.com/odoo/odoo/blob/2a08ba3ac193965e3394bb4e75d7e3abd39e68d3/addons/product/models/product_category.py#L46

Reply:
I made functional tests and recursions seem to be under control.

Reply:
But the tests are only executed when Odoo is started so that it can execute them, but this verification is NEVER performed until this is the case and it can cause problems. If the system has a revision we can take advantage of it.